lib/sycamore/sycamore/data/document.py

```diff
@@ -328,6 +329,51 @@ def split_data_metadata(all: list[Document]) -> tuple[list[Document], list[Metad
     )
 
 
+class SummarizeDocument(Document):
+    def __init__(self, document=None, **kwargs):
+        if "elements" in kwargs:
+            raise ValueError("Cannot set elements directly in a SummarizeDocument")
+        super().__init__(document, **kwargs)
+        if self.data.get("sub_docs") is None:
+            self.data["sub_docs"] = []
+        elif not isinstance(sd := self.data["sub_docs"], list):
+            raise ValueError(f"sub_docs must be a list of Document, found {sd}")
+        else:
+            subdocs = self.data["sub_docs"]
+            for sd in subdocs:
+                if not isinstance(sd, (dict, UserDict, Document)):
```

Comment:
Is this for convenience?  I would think you could force the caller to make everything a Document.